### PR TITLE
chore(tests): clean up test suite

### PR DESCRIPTION
**Pull request by jniles**
This commit refactors the test suite to promote rigor and speed.  The following changes are added:
1. Remove all locations, except for the default enterprise location.
2. Prefer using Notify.hasDanger() over custom error messages.
3. Use `bhConstants` for min/max dates.
4. Ensure that elements selected in lists are unique, and protractor will never have to resolve those elements stochastically.
5. When a Notify.has*() call fails, it echos a message so developers  know what happened.

There have been some flaky test cases in master.  Hopefully this commit helps address those.

This commit also cleans up our end-to-end tests by ensuring that every single end to end test selects only a single element on the screen.  This means there will no longer be any warnings related to randomly selecting the first element of a group.

Finally, locations are precisely specified in the test file using FU.select() controls.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
@sfount, could I get a review?

**Comment by jniles** [comment]
Ping @sfount.

File: client/src/js/app.js [hunk 1]
```diff
@@ -569,7 +569,14 @@ function constantConfig() {
     },
     purchase : {
       GRID_HEIGHT: 200
-    }
+    },
+    dates : {
```
Comment (sfount) [inline client/src/js/app.js:573]: :+1:

**Comment by sfount** [comment]
Excellent re-factor :100: LGTM.

**Pull request merged**